lib/b2b/idp.ts

```diff
@@ -0,0 +1,205 @@
+  expires_at: number;
+  issued_at: number;
+  issuer: string;
+  not_before: number;
+  token_type: string;
+}
+
+export class IDP {
+  private fetchConfig: fetchConfig;
+  private jwtConfig: JwtConfig;
+  private jwksClient: jose.JWTVerifyGetKey;
+  private policyCache: PolicyCache;
+
+  constructor(
+    fetchConfig: fetchConfig,
+    jwtConfig: JwtConfig,
+    policyCache: PolicyCache
+  ) {
+    this.fetchConfig = fetchConfig;
+    this.jwtConfig = jwtConfig;
+    this.jwksClient = jwtConfig.jwks;
+    this.policyCache = policyCache;
+  }
+
+  async introspectTokenNetwork(
+    data: IntrospectTokenRequest,
+    options?: {
+      authorization_check?: AuthorizationCheck;
+    }
+  ): Promise<IntrospectTokenClaims> {
+    const fetchConfig: fetchConfig = {
+      ...this.fetchConfig,
+      headers: {
+        ["User-Agent"]: this.fetchConfig.headers["User-Agent"],
+        "Content-Type": "application/x-www-form-urlencoded",
+      },
+    };
+
+    const params: Record<string, string> = {
+      token: data.token,
+      client_id: data.client_id,
+    };
+
+    if (data.client_secret && data.client_secret.length > 0) {
+      params.client_secret = data.client_secret;
+    }
+
+    if (data.token_type_hint && data.token_type_hint.length > 0) {
+      params.token_type_hint = data.token_type_hint;
+    }
+
+    let response;
+    try {
+      response = await request<IntrospectTokenResponse>(fetchConfig, {
+        method: "POST",
+        url: `/v1/public/${this.jwtConfig.projectID}/oauth2/introspect`,
+        dataRaw: new URLSearchParams(params),
+      });
+    } catch (err) {
+      throw new ClientError("token_invalid", "Could not introspect token", err);
+    }
+    const {
+      /* eslint-disable @typescript-eslint/no-unused-vars */
+      aud: _aud,
+      exp: _exp,
+      iat: _iat,
+      iss: _iss,
+      nbf: _nbf,
+      sub: _sub,
+      status_code: _status_code,
+      scope: _scope,
+      active: _active,
+      request_id: _request_id,
+      token_type: _token_type,
+      client_id: _client_id,
+      /* eslint-enable @typescript-eslint/no-unused-vars */
+      ...customClaims
+    } = response;
+
+    if (!_active) {
+      throw new ClientError("token_invalid", "Token was not active", null);
+    }
+
+    if (options?.authorization_check) {
+      const policy = await this.policyCache.getPolicy();
+      performScopeAuthorizationCheck({
+        policy,
+        tokenScopes: (_scope as string).trim().split(" "),
+        authorizationCheck: options.authorization_check,
+      });
+    }
+
+    return {
+      subject: _sub as string,
+      scope: _scope as string,
+      audience: _aud as string[],
+      expires_at: _exp as number,
+      issued_at: _iat as number,
+      issuer: _iss as string,
+      not_before: _nbf as number,
+      custom_claims: customClaims,
+      token_type: _token_type as string,
+    };
+  }
+
+  async introspectTokenLocal(
+    data: IntrospectTokenRequest,
```

Comment:
we no longer need to take in a `client_id`, `client_secret`, or `token_type_hint` here